bindings/go/runtime/json_schema_generator_test.go

```diff
@@ -0,0 +1,58 @@
+package runtime
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type customType struct {
+	Type            Type   `json:"type"`
+	AdditionalField string `json:"additionalField"`
+}
+
+func (c *customType) GetType() Type {
+	return c.Type
+}
+
+func (c *customType) SetType(t Type) {
+	c.Type = t
+}
+
+func (c *customType) DeepCopyTyped() Typed {
+	c2 := *c
+	return &c2
+}
+
+var _ Typed = &customType{}
+
+func TestGenerateJSONSchemaForType(t *testing.T) {
```

Comment:
Im missing a failure case

Reply:
Plus the function requires to pass in an struct that satisfies the interface. So I can't pass garbage.